sympy/combinatorics/coset_table.py

```diff
@@ -831,41 +827,26 @@ def modified_scan(self, alpha, w, y, fill=False):
     def modified_scan_and_fill(self, alpha, w, y):
         self.modified_scan(alpha, w, y, fill=True)
 
-    def modified_rep(self, k, p_p):
+    def modified_rep(self, k):
         '''
         Input: C, κ∈Ω, , pP
         '''
-        p = self.p
-        lambda_ = k
-        rho = p[lambda_]
-        # Array to trace back compression path.
-        s = p[:]
-        while rho != lambda_:
-            s[rho] = lambda_
-            lambda_ = rho
-            rho = p[lambda_]
-        rho = s[lamda]
-        while rho != k:
-            mu = rho
-            rho = s[mu]
-            p[rho] = lambda_
-            p_p[rho] = p_p[rho]*p_p[mu]
-        return lambda_
-
-    def modified_merge(k, lamda, w, p_p, q, l):
+        self.rep(k, modified=True)
+
+    def modified_merge(k, lamda, w, q):
         '''
         Input: κ, λ∈Ω, w∈(Y∪Y–1)*withτ(κ)= φ(w)τ(λ), p_p, q,l GP
         q is a queue of length l of elements to be deleted from Ω *
         '''
-        phi = self.modified_rep(k, p_p)
-        psi = self.modified_rep(lamda, p_p)
+        phi = self.modified_rep(k)
+        psi = self.modified_rep(lamda)
         if phi > psi:
             self.p[phi] = psi
-            p_p[phi] =  p_p[k]**-1*w*p_p[lamda]
+            self.p_p[phi] =  self.p_p[k]**-1*w*self.p_p[lamda]
             q.append(phi)
         elif phi < psi:
             self.p[psi] = phi
-            p_p[psi] = p_p[lamda]**-1*w**-1*p_p[k]
+            self.p_p[psi] = self.p_p[lamda]**-1*w**-1*self.p_p[k]
             q.append(psi)
```

Comment:
Did you see how it's written out in regular `merge`? You can use `min(phi, psi)` and make this twice as short.